capstone-frontend/src/components/LeftSideBar.jsx

```diff
@@ -1,44 +1,158 @@
-        </Link>
-        <Link to="/browse" className="bg-dark list-group-item list-group-item-action">
-          <div className="d-flex w-100 justify-content-start align-items-center">
-            <span className="fa fa-tasks fa-fw mr-3"></span>
-            <span id="browse-link" className="menu-collapsed"> Browse </span>
-          </div>
-        </Link>
-        <Link to="/myreads" className="bg-dark list-group-item list-group-item-action">
-          <div className="d-flex w-100 justify-content-start align-items-center">
-            <span className="fa fa-tasks fa-fw mr-3"></span>
-            <span id="mybooks-link" className="menu-collapsed"> My Books </span>
-          </div>
-        </Link>
-        <Link to="/mylists" className="bg-dark list-group-item list-group-item-action">
-          <div className="d-flex w-100 justify-content-start align-items-center">
-            <span className="fa fa-tasks fa-fw mr-3"></span>
-            <span id="mylists-link" className="menu-collapsed"> My Lists </span>
-          </div>
-        </Link>
-        <Link to="/myclubs" className="bg-dark list-group-item list-group-item-action">
-          <div className="d-flex w-100 justify-content-start align-items-center">
-            <span className="fa fa-tasks fa-fw mr-3"></span>
-            <span id="myclubs-link" className="menu-collapsed"> My Clubs </span>
-          </div>
-        </Link>        
-      </ul>
-    </div>
-  );
-};
-
-export default LeftSideBar;
\ No newline at end of file
+import { Accordion, Card, Button} from 'react-bootstrap';
+import "../styles/LeftSideBar.css";
+
+
+export class LeftSideBar extends Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      bookLists: []
+    }
+  }
+
+  fetchBookLists = async () => {
+
+    const userID = window.localStorage.getItem("userID");
+
+    console.log("Fetching Books LeftSideBar");
+
+    const bookLists = await fetch(`/api/booklist?userID=${userID}`, {
+      method: "GET",
+    }).then(resp => resp.json());
+
+    this.setState({ bookLists });
+  }
+
+  componentDidMount() {
+    this.fetchBookLists();
+  }
+
+  deleteBookList = async (bookListId) => {
+    console.log("DELETE BookList ID:\t", bookListId);
+
+    const bookListJson = {
+      "bookListID": bookListId
+    }
+
+    // Delete BookList in Firebase
+    await fetch("/api/booklist", {
+      method: "DELETE",
+      body: JSON.stringify(bookListJson)
+    });
+
+    this.fetchBookLists();
+
+    const currentUrlPath = new URL(window.location.href).pathname;
+    const bookListIdURL = currentUrlPath.substr(10);
+
+    if (bookListIdURL === bookListId) {
+      document.location.href="/";
+    }
+  }
+
+
+  render() {
+    return (
+      <div id="left-sidebar-container" className="sidebar-expanded d-none d-md-block col-2 sidebar-container">
+        <ul className="list-group sticky-top sticky-offset">
+
+          <Accordion defaultActiveKey="0" style={{ marginRight: -14.3 }}>
```

Comment:
How was the number -14.3 decided upon? Will we have issues resizing to different sized devices with hardcoded values like this?

Reply:
The ```marginRight``` code makes sure the LeftSideBar is the same size as background for the LeftSideBar. It doesn't affect the smaller devices. We can change the style later as well, if we want to redesign the sidebar.